hera_cal/lstbin.py

```diff
@@ -175,13 +212,39 @@ def lst_bin(data_list, lst_list, flags_list=None, dlst=None, begin_lst=None, lst
                     if ind not in data[key]:
                         data[key][ind] = []
                         flags[key][ind] = []
+                        nsamples[key][ind] = []
                     # append data ndarray to LST bin
                     data[key][ind].append(d[key][k])
                     # also insert flags if fed
                     if flags_list is None:
                         flags[key][ind].append(np.zeros_like(d[key][k], np.bool))
                     else:
                         flags[key][ind].append(flags_list[i][key][k])
+                    if nsamples_list is None:
+                        nsamples[key][ind].append(np.ones_like(d[key][k], np.int))
+                    else:
+                        nsamples[key][ind].append(nsamples_list[i][key][k])
+        # add in spoofed baselines to keep baselines in different LST files consistent.
+        if bl_list is not None:
+            for antpair in bl_list:
+                for pol in pols:
+                    key = antpair + (pol,)
+                    if key not in data:
+                        if (key[0] != key[1] and utils.reverse_bl(key) not in data) or key[0] == key[1]:
+                            # last part lets us spoof ne and en for autocorrs.
+                            nsamples[key] = odict({ind: [] for ind in range(len(lst_grid))})
+                            data[key] = odict({ind: [] for ind in range(len(lst_grid))})
+                            flags[key] = odict({ind: [] for ind in range(len(lst_grid))})
+
+                    for ind in range(len(lst_grid)):
```

Comment:
Since different nights have different sets of baselines and different LST bins have different sets of nights, it is possible to get a baseline that appears in a subset of the LSTs within an LST chunk (for example, a baseline that exists in one of the nights that only contained a subset of the LSTs in the LST chunk being processed). I've added a comment explaining this.